### PR TITLE
feat: prevent duplicate email in group members

### PR DESCRIPTION
**Pull request by lukaszcz**
## Summary
This PR prevents duplicate email addresses in group members by adding database constraints and UI validation.

## Changes
- Add UNIQUE constraint on `group_members(group_id, connected_user_id)` (allows NULL)
- Add `isDuplicateMemberEmail()` validation function
- Update add-member and edit-member screens to validate duplicate emails
- Update `KnownUserSuggestionInput` component to show duplicate email errors
- Add comprehensive tests for email and name duplicate validation

## Testing
- Added 17 new test cases covering both name and email duplicate detection
- Tests cover edge cases like case-insensitivity, empty values, and undefined values

Closes #79

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: supabase/migrations/20260215173237_add_unique_connected_user_id_to_group_members.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+-- Add UNIQUE constraint to connected_user_id column in group_members table
+-- This prevents the same user from being added to the same group multiple times
+-- NULL values are allowed (for unconnected/invited members), but all non-NULL values must be unique per group
+
+-- First, we need to add a partial unique index that only applies to non-NULL connected_user_id values
+-- and is scoped to each group (a user can be in multiple groups, just not the same group twice)
+CREATE UNIQUE INDEX idx_group_members_unique_connected_user
+ON public.group_members (group_id, connected_user_id)
+WHERE connected_user_id IS NOT NULL;
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply supabase/migrations/20260215173237_add_unique_connected_user_id_to_group_members.sql:9]: That's okay, the database hasn't been deployed to production yet.
